### ISSUE TITLE
'data' referenced before assignment

### ISSUE THREAD

**Issue by garretsimpson** [Yelp/elastalert#1372]
https://github.com/Yelp/elastalert/blob/2e96c34b07bd9b5dafdde130dfe1722a1c5e99a8/elastalert/create_index.py#L77
```
  File "/usr/lib/python2.7/site-packages/elastalert/create_index.py", line 77, in main
    username = args.username if args.username else data.get('es_username')
UnboundLocalError: local variable 'data' referenced before assignment
```

**Comment by windowsrefund** [comment]
Also seeing this

**Comment by liulongfei** [comment]
Please add the following in line 76 below:
```
        data = {}
```


After adding the effect is as follows:
```
    else:
        data = {}
        username = args.username if args.username else data.get('es_username')
        password = args.password if args.password else data.get('es_password')
```

For me it is working and good luck.

**Comment by Qmando** [comment]
Looks like this occurs if you don't have a config.yaml and don't specify host/port as arguments. I'll fix this today.

**Comment by Qmando** [comment]
Fixed in #1382
